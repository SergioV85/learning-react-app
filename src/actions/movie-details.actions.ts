import * as MovieDetailsActionTypes from './../action-types/movie-details.action-types';
import { getMovieDetails } from './../services/movies.service';

const backToListAction = () => ({
  type: MovieDetailsActionTypes.MovieDetailsActionTypes.BackToList,
});
const getMovie = (selectedMovieId: number) => {
  return (dispatch: any) => {
    dispatch({
      payload: selectedMovieId,
      type: MovieDetailsActionTypes.MovieDetailsActionTypes.GetMovie,
    });

    return getMovieDetails(selectedMovieId)
      .then((details) => dispatch({
        payload: details.data,
        type: MovieDetailsActionTypes.MovieDetailsActionTypes.GetMovieComplete,
      }))
      .catch((err) => dispatch({
        payload: err,
        type: MovieDetailsActionTypes.MovieDetailsActionTypes.GetMovieError,
      }));
  };
};

export const MovieDetailsActions = {
  backToListAction,
  getMovie,
};
