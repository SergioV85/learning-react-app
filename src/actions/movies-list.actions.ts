import * as MoviesListActionTypes from './../action-types/movies-list.action-types';
import { getMoviesList } from './../services/movies.service';

const searchMovies = () => {
  return (dispatch: any, getState: any) => {
    const { searchParams } = getState();
    dispatch({
      payload: searchParams,
      type: MoviesListActionTypes.MoviesListActionTypes.GetMovies,
    });

    return getMoviesList(searchParams)
      .then((list) => dispatch({
        payload: list.data,
        type: MoviesListActionTypes.MoviesListActionTypes.GetMoviesComplete,
      }))
      .catch((err) => dispatch({
        payload: err,
        type: MoviesListActionTypes.MoviesListActionTypes.GetMoviesError,
      }));
  };
};

export const MoviesListActions = {
  searchMovies,
};
