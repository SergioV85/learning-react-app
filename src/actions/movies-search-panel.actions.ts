import * as MoviesSearchPanelActionTypes from './../action-types/movies-search-panel.action-types';

const updateKeyword = (keyword: string) => ({
  payload: keyword,
  type: MoviesSearchPanelActionTypes.MoviesSearchPanelActionTypes.UpdateSearchKeyword,
});
const updateSortingOrder = (sortOrder: string) => ({
  payload: sortOrder,
  type: MoviesSearchPanelActionTypes.MoviesSearchPanelActionTypes.UpdateSortOrder,
});
const updateSortingType = (sortType: string) => ({
  payload: sortType,
  type: MoviesSearchPanelActionTypes.MoviesSearchPanelActionTypes.UpdateSortType,
});
const updateType = (searchType: string) => ({
  payload: searchType,
  type: MoviesSearchPanelActionTypes.MoviesSearchPanelActionTypes.UpdateSearchType,
});

export const MoviesSearchPanelActions = {
  updateKeyword,
  updateSortingOrder,
  updateSortingType,
  updateType,
};
