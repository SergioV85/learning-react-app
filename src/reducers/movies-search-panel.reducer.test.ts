import { MoviesSearchPanelActions } from './../actions/movies-search-panel.actions';
import { moviesSearchPanelReducer } from './movies-search-panel.reducer';

const mockedState = {
  search: '',
  searchBy: 'title',
  sortBy: 'rating',
  sortOrder: 'desc',
};

const initialState = moviesSearchPanelReducer(mockedState, {} as any);

describe('Movies Search Panel Reducer', () => {
  describe('initial state', () => {
    it('should match a snapshot', () => {
      expect(initialState).toMatchSnapshot();
    });
  });
});
