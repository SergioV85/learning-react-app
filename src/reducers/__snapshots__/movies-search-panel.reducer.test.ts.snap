// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Search Panel Reducer initial state should match a snapshot 1`] = `
Object {
  "search": "",
  "searchBy": "title",
  "sortBy": "rating",
  "sortOrder": "desc",
}
`;
